src/bindings/Next.res

```diff
@@ -0,0 +1,141 @@
+module GetServerSideProps = {
+  module Req = {
+    type t
+  }
+
+  module Res = {
+    type t
+
+    @send external setHeader: (t, string, string) => unit = "setHeader"
+    @send external write: (t, string) => unit = "write"
+    @send external end_: t => unit = "end"
+  }
+
+  // See: https://github.com/zeit/next.js/blob/canary/packages/next/types/index.d.ts
+  type context<'props, 'params> = {
+    params: Js.t<'params>,
+    query: Js.Dict.t<string>,
+    req: Req.t,
+    res: Res.t,
+  }
+
+  type t<'props, 'params> = context<'props, 'params> => Js.Promise.t<{"props": 'props}>
+}
+
+module GetStaticProps = {
+  // See: https://github.com/zeit/next.js/blob/canary/packages/next/types/index.d.ts
+  type context<'props, 'params, 'previewData> = {
+    params: 'params,
+    preview: option<bool>, // preview is true if the page is in the preview mode and undefined otherwise.
+    previewData: Js.Nullable.t<'previewData>,
+    query: Js.Dict.t<string>,
+    req: Js.Nullable.t<Js.t<'props>>,
+  }
```

Comment:
Query and req dont exist. My best guess is these are hearing from copying getinitialprops bindings before that was deprecated

Reply:
ah yes, I remembered that someone made a comment about that, but I couldn't find the place where someone made this observation... probably in some post in the forum.

Will fix this on master